data/transactions/stateproof.go

```diff
@@ -42,6 +45,31 @@ func (sp StateProofTxnFields) Empty() bool {
 		sp.Message.MsgIsZero()
 }
 
+// String returns a string representation of the StateProofTxnFields.
+func (sp StateProofTxnFields) String() string {
+	return fmt.Sprintf(
```

Comment:
I really like the content of the structure - i.e what we choose to print.
few points
- should make sure a field is empty before printing
- should be in one line
- I think we should give up the "keys" just to be aligned with how transaction is being printed.